Userland/Libraries/LibGfx/Bitmap.cpp

```diff
@@ -633,4 +635,61 @@ Optional<Color> Bitmap::solid_color(u8 alpha_threshold) const
     return color;
 }
 
+static float color_distance_squared(Gfx::Color const& lhs, Gfx::Color const& rhs)
+{
+    int a = rhs.red() - lhs.red();
+    int b = rhs.green() - lhs.green();
+    int c = rhs.blue() - lhs.blue();
+    int d = rhs.alpha() - lhs.alpha();
+    return (a * a + b * b + c * c + d * d) / (4.0f * 255.0f * 255.0f);
+}
+
+bool Bitmap::can_paint(Gfx::IntPoint point, Gfx::Color const& target_color, float threshold_normalized_squared) const
+{
+    auto pixel_color = get_pixel<Gfx::StorageFormat::BGRA8888>(point.x(), point.y());
+    return color_distance_squared(pixel_color, target_color) <= threshold_normalized_squared;
+}
```

Comment:
If feels a bit weird making this a method on the bitmap since it seems an implementation detail of the flood fill (and the method does not seem to have any use outside of that).

I'd make something like this a lambda within the `flood_visit` method, or just do the check inline (does this even need to be a function?).

Reply:
Completely agree now that you mention it. No real reason for it to be a function, this is leftover from my initial implementation in Bucket Tool before it used flood fill. I've fixed this.